wwwbasic.js

```diff
@@ -1482,32 +1493,39 @@
         if (m = cmds.match(/^(u|d|l|r|e|f|g|h|a|ta|c|s)([0-9]+)?/)) {
           var op = m[1];
           var n = m[2] == '' ? 1 : parseInt(m[2]);
+          if(Math.sign(n) )
           if (op == 'c') {
             draw_state.color = n;
           } else if (op == 'a') {
-            draw_state.angle = n;
-            // TODO: Implement.
+            if(n >= 0 && n <= 3){
+              draw_state.angle = n * 90;
+            } else {
+              Throw("Invalid A Rotation");
+            }
           } else if (op == 'ta') {
-            draw_state.turn_angle = n;
-            // TODO: Implement.
+            if(n > -360 && n < 360){
+              draw_state.angle = n;
+            } else {
+              Throw("Invalid TA Rotation");
+            }
           } else if (op == 's') {
             draw_state.scale = n / 4;
           } else if (op == 'u') {
             Step(0, -n);
           } else if (op == 'd') {
             Step(0, n);
           } else if (op == 'l') {
-            Step(-n, 0);
+            Step(-n,0);
           } else if (op == 'r') {
-            Step(n, 0);
+            Step(n,0);
           } else if (op == 'e') {
-            Step(n, -n);
+            Step(rotate_x(n,135), rotate_y(-n,135));
```

Comment:
Couldn't the rotate live inside Step ()?

Reply:
Not unless we added a third parameter to the Step function because the U L D R uses something besides rotate_x(n,o)/rotate_y(n,o) because rotate_x/rotate_y(0,?)==0 and the 0 needs to shift, which is the reason for the o parameter.